internal/controller/api/v1beta1/agentserviceconfig_types.go

```diff
@@ -22,6 +22,18 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// OSImage defines an Operating System image and the OpenShift version it
+// is associated with.
+type OSImage struct {
+	// OpenshiftVersion is the Major.Minor version of OpenShift that this image
+	// is to be associated with.
+	OpenshiftVersion string `json:"openshiftVersion"`
+	// Version is the Operating System version of the image.
+	Version          string `json:"version"`
+	// Url specifies the path to the Operating System image.
+	Url              string `json:"url"`
```

Comment:
Is this the URL for the ISO?
If so, maybe the name should mention it?
Remember that we also need the url for the rootfs

Reply:
For the rootfs url https://issues.redhat.com/browse/MGMT-5429

Reply:
I think `OSImage.Url` is clear for the general case of providing the image. For the case where the rootfs portion is being provided separately, we can add `OSImage.RootFSUrl` or similar in the next task.